R/pkg/R/mllib.R

```diff
@@ -694,8 +694,14 @@ setMethod("predict", signature(object = "KMeansModel"),
 #' }
 #' @note spark.mlp since 2.1.0
 setMethod("spark.mlp", signature(data = "SparkDataFrame"),
-          function(data, blockSize = 128, layers = c(3, 5, 2), solver = "l-bfgs", maxIter = 100,
-                   tol = 0.5, stepSize = 1, seed = 1) {
+          function(data, layers, blockSize = 128, solver = "l-bfgs", maxIter = 100,
+                   tol = 1E-6, stepSize = 0.03, seed = 0x7FFFFFFF) {
+            if (length(layers) <= 1) {
+              stop("layers vector require length > 0.")
+            }
+            if (any(sapply(layers, function(e) !is.numeric(e)))) {
```

Comment:
You can use `numToInt` from https://github.com/apache/spark/blob/master/R/pkg/R/utils.R#L368 -- It'll print a warning if its not an integer